src/lib/do.py

```diff
@@ -60,16 +77,21 @@ def check_uptime(self, max_uptime):
         :param max_uptime: int. Uptime threshold in hours
         :return: lst. hosts violating `max_uptime`.
         """
+
         hosts_violated = []
+        droplets = self.c.get_all_droplets()
         try:
-            for i in self.c.get_uptime():
-                log.debug('checking uptime of host {0}'.format(i))
+            for i in droplets:
+                log.debug('checking uptime of instance {0}'.format(i))
                 # ts format from digital ocean api: 2015-05-07T22:27:38Z
-                created_at = dateutil.parser.parse(i['created_at'])
-                diff = relativedelta(datetime.now(), created_at)
-                if (diff.days * 24) + diff.hours > max_uptime:
+                created_at = i.created_at
+                log.debug('instance {0} was created at {1}'.format(i, created_at))
+                # slice off last char of time stamp to make it timezone unaware.
+                created_at = dateutil.parser.parse(created_at[:-1])
+                if self.eval_delta(created_at,datetime.utcnow(), max_uptime):
                     hosts_violated.append(i)
-        except:
+        except Exception as e:
+            log.exception(e)
```

Comment:
I'm sure you're aware, but by doing this the caller loses all information about what kind of exception it was.  I assume you're ok with that. Also keep in mind that this will catch things like SyntaxError which may be annoying. It's usually better to catch specific exceptions if you can reasonably do so.